app/components/views/GetStartedPage/CreateWallet/index.js

```diff
@@ -1,82 +1,149 @@
 import { createElement as h } from "react";
 import { DecredLoading } from "indicators";
+import { useState, useEffect, useCallback } from "react";
+import { useService } from "@xstate/react";
+import { useSelector, useDispatch } from "react-redux";
+import * as wla from "actions/WalletLoaderActions";
+import * as sel from "selectors";
+import { sendParent } from "xstate";
 
-@autobind
-class CreateWallet extends React.Component {
-  service;
-  constructor(props) {
-    super(props);
-    const { sendEvent, sendContinue, checkIsValid, onCreateWatchOnly } = this;
-    const { backToCredentials, cancelCreateWallet, generateSeed } = props;
-    this.machine = CreateWalletMachine({
-      generateSeed,
-      backToCredentials,
-      cancelCreateWallet,
-      sendEvent,
-      sendContinue,
-      checkIsValid,
-      onCreateWatchOnly
-    });
-    this.service = interpret(this.machine).onTransition((current) =>
-      this.setState({ current }, this.getStateComponent)
-    );
-    this.state = {
-      current: this.machine.initialState,
-      StateComponent: null,
-      isValid: false
-    };
-  }
-
-  componentDidMount() {
-    this.service.start();
-    const isNew = !this.props.createWalletExisting;
-    const isWatchingOnly = this.props.isCreatingWatchingOnly;
-    // TODO Add watching only state and tezos
-    const { isTrezor, isTestNet } = this.props;
-    this.service.send({ type: "CREATE_WALLET", isNew, isTestNet });
-    this.service.send({ type: "RESTORE_WATCHING_ONLY_WALLET", isWatchingOnly });
-    this.service.send({ type: "RESTORE_TREZOR_WALLET", isTrezor });
-    this.service.send({
-      type: "RESTORE_WALLET",
-      isRestore: !isNew,
-      isWatchingOnly
-    });
-  }
-
-  componentWillUnmount() {
-    this.service.stop();
-  }
-
-  getStateComponent() {
-    const { current, isValid } = this.state;
-    const {
-      sendBack,
-      sendContinue,
-      setPassPhrase,
-      setSeed,
-      setError,
-      onCreateWallet
-    } = this;
-    const { mnemonic, error } = this.service._state.context;
-    const { decodeSeed } = this.props;
-    let component, text;
+const CreateWallet = ({ createWalletRef }) => {
+  const dispatch = useDispatch();
+  const decodeSeed = useCallback((seed) => dispatch(wla.decodeSeed(seed)), [
+    dispatch
+  ]);
+  const cancelCreateWallet = useCallback(
+    () => dispatch(wla.cancelCreateWallet()),
+    [dispatch]
+  );
+  const generateSeed = useCallback(() => dispatch(wla.generateSeed()), [
+    dispatch
+  ]);
+  // TODO implement pubpass
+  const createWatchOnlyWalletRequest = useCallback(
+    (extendedPubKey, pubPass = "") =>
+      dispatch(wla.createWatchOnlyWalletRequest(extendedPubKey, pubPass)),
+    [dispatch]
+  );
+  const createWalletRequest = useCallback(
+    (pubpass, passPhrase, seed, isNew) =>
+      dispatch(wla.createWalletRequest(pubpass, passPhrase, seed, isNew)),
+    [dispatch]
+  );
+  const isTestNet = useSelector(sel.isTestNet);
```

Comment:
Can we consider creating custom hook and move all state/redux related logic? 
This component is pretty big, and it may improve the readability.